pkg/prometheus/operator.go

```diff
@@ -1159,28 +1160,60 @@ func loadAdditionalScrapeConfigsSecret(additionalScrapeConfigs *v1.SecretKeySele
 	return nil, nil
 }
 
+func extractCredKey(secret *v1.Secret, sel v1.SecretKeySelector, cred string) (string, error) {
+	if s, ok := secret.Data[sel.Key]; ok {
+		return string(s), nil
+	} else {
```

Comment:
nit: there is no need for this `else` here since you are returning early

Reply:
updated